src/main/java/ee/carlrobert/codegpt/completions/CompletionRequestProvider.java

```diff
@@ -124,13 +136,62 @@ public LlamaCompletionRequest buildLlamaCompletionRequest(
         message.getPrompt(),
         conversation.getMessages());
     var configuration = ConfigurationState.getInstance();
+    LlamaRequestSettings llamaRequestSettings = settings.getRequestSettings();
     return new LlamaCompletionRequest.Builder(prompt)
         .setN_predict(configuration.getMaxTokens())
         .setTemperature(configuration.getTemperature())
-        .setTop_k(settings.getTopK())
-        .setTop_p(settings.getTopP())
-        .setMin_p(settings.getMinP())
-        .setRepeat_penalty(settings.getRepeatPenalty())
+        .setTop_k(llamaRequestSettings.getTopK())
+        .setTop_p(llamaRequestSettings.getTopP())
+        .setMin_p(llamaRequestSettings.getMinP())
+        .setRepeat_penalty(llamaRequestSettings.getRepeatPenalty())
+        .build();
+  }
+
+  public OllamaCompletionRequest buildOLlamaCompletionRequest(
+      Message message,
+      ConversationType conversationType,
+      boolean isStream) {
+    var settings = OllamaSettingsState.getInstance();
+    PromptTemplate promptTemplate;
+    String model;
+    if (settings.isRunLocalServer()) {
+      var localSettings = settings.getLocalSettings();
+      var localModel = localSettings.getModel();
+      model = LlamaCompletionModel.getOllamaId(localModel);
+      if (localModel instanceof CustomLlamaModel) {
+        promptTemplate = localSettings.getChatPromptTemplate();
+      } else {
+        promptTemplate = LlamaModel.findByHuggingFaceModel((HuggingFaceModel) localModel)
+            .getPromptTemplate();
+      }
+    } else {
+      var remoteSettings = settings.getRemoteSettings();
+      promptTemplate = remoteSettings.getChatPromptTemplate();
+      model = LlamaCompletionModel.getOllamaId(remoteSettings.getModel());
+    }
+
+    var systemPrompt = COMPLETION_SYSTEM_PROMPT;
+    if (conversationType == ConversationType.FIX_COMPILE_ERRORS) {
+      systemPrompt = FIX_COMPILE_ERRORS_SYSTEM_PROMPT;
+    }
+
+    var prompt = promptTemplate.buildPrompt(
+        "",
+        message.getPrompt(),
+        List.of());
+    var configuration = ConfigurationState.getInstance();
+    var requestSettings = settings.getRequestSettings();
+    return new OllamaCompletionRequest.Builder(model, prompt)
+        .setOptions(new OllamaParameters(new Builder()
+            .temperature(configuration.getTemperature())
+            .numPredict(configuration.getMaxTokens())
+            .topK(requestSettings.getTopK())
+            .topP(requestSettings.getTopP())
+            .repeatPenalty(requestSettings.getRepeatPenalty())
+        ))
+        .setSystem(systemPrompt)
+//        .setTemplate(promptTemplate.toString())
```

Comment:
The `template` for custom promp templates is defined by: [ollama/modelfile.md](https://github.com/ollama/ollama/blob/main/docs/modelfile.md#template)